core/templates/dev/head/exploration_editor/RouterServices.js

```diff
@@ -98,6 +83,30 @@ oppia.factory('routerService', [
       }
     });
 
+    var _doNavigation = function(path, pathType) {
```

Comment:
Maybe call this _doNavigationWithState, so that devs don't think it's a general purpose navigation function to be used for /settings, /history, etc.